src/Styleguide/Elements/Checkbox.tsx

```diff
@@ -49,7 +50,13 @@ export class Checkbox extends React.Component<CheckboxProps, CheckboxState> {
     const { children } = this.props
 
     return (
-      <Flex my={0.5} onClick={this.toggleSelected}>
+      <Flex
+        my={0.5}
+        onClick={() => {
+          this.props.onSelect && this.props.onSelect(this.state.selected)
+          this.toggleSelected
+        }}
```

Comment:
In a follow up PR we can consolidate / clean up and move this stuff into a handler on the class.